runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkExecutableStageContext.java

```diff
@@ -37,11 +43,18 @@
   }
 
   public static Factory batchFactory() {
-    return null;
+    return jobInfo -> {
```

Comment:
This "factory" should manage singleton state in this case. We want to make sure there's only a single instance of a DockerJobBundleFactory for a given job id. In order to do so, the factory itself needs to be a singleton.